docs/UpgradeGuide2.md

```diff
@@ -0,0 +1,529 @@
+# Upgrading ddtrace to 2.0
+
+## From 1.x to 2.0
+
+Upgrading `ddtrace` from 1.x to 2.0 introduces some breaking changes which are outlined below.
+
+[**Upgrade basic usage**](#2.0-upgrading-basic-usage)
+
+For users with a basic implementation (configuration file + out-of-the-box instrumentation), only minor changes to your configuration file are required: most applications take just minutes to update. Check out the following sections for a step-by-step guide.
+
+- [Requires Ruby 2.5+](#2.0-requires-ruby-2.5+)
+- [Extracts datadog-ci gem](#2.0-extracts-ci)
+- [Configuration changes](#2.0-configuration-changes)
+  - [Replace `use` with `instrument`](#2.0-use-instrument)
+  - [Type checking](#2.0-type-checking)
+  - [Propagation default](#2.0-propagation-default)
+  - [Options](#2.0-options)
+
+---
+
+[**Upgrade advanced usage**](#2.0-upgrading-advanced-usage)
+
+For users with an advanced implementation of `ddtrace` (custom instrumentation, sampling or processing behavior, etc), some additional changes may be required. See the following sections for details about what features changed and how to use them.
+
+- [Frozen String](#2.0-frozen-string)
+- [Tracing API](#2.0-tracing-api)
+- [Configuration API](#2.0-configuration-api)
+- [Log Correlation](#2.0-log-correlation)
+- [Distributed Tracing](#2.0-distributed-tracing)
+- [Transport](#2.0-transport)
+- [Sampling](#2.0-sampling)
+
+---
+
+[**Upgrade Instrumentations**](#2.0-upgrade-instrumentations)
```

Comment:
We use the word upgrade in several of the headlines like `Upgrade advanced usage` and `Upgrade basic usage`. Is that a bit redundant since this is the upgrade guide? Should we just drop the word `upgrade` from these sections?